hasher-matcher-actioner/hmalib/common/signal_models.py

```diff
@@ -123,35 +124,53 @@ def get_from_signal(
         signal_id: t.Union[str, int],
         signal_source: str,
     ) -> t.List["PDQSignalMetadata"]:
-
+        """
+        Load the metadata objects relaed to this signal.
+        Optionally provide a data set id to filter to
+        """
         items = table.query(
             KeyConditionExpression=Key("PK").eq(
                 cls.get_dynamodb_signal_key(signal_source, signal_id)
             )
             & Key("SK").begins_with(cls.DATASET_PREFIX),
-            ProjectionExpression="PK, ContentHash, UpdatedAt, SK, SignalSource, SignalHash, Tags, PendingOpinionChange",
             FilterExpression=Attr("HashType").eq(cls.SIGNAL_TYPE),
+            ProjectionExpression=cls.SIGNAL_TYPE_PROJECTION_EXPRESSION,
         ).get("Items", [])
         return cls._result_items_to_metadata(items)
 
+    @classmethod
+    def get_from_signal_and_ds_id(
+        cls, table: Table, signal_id: t.Union[str, int], signal_source: str, ds_id: str
+    ) -> t.Optional["PDQSignalMetadata"]:
+        item = table.get_item(
+            Key={
+                "PK": cls.get_dynamodb_signal_key(signal_source, signal_id),
+                "SK": cls.DATASET_PREFIX + ds_id,
+            },
+            ProjectionExpression=cls.SIGNAL_TYPE_PROJECTION_EXPRESSION,
```

Comment:
get_item doesn't need this if you want all the fields anyway.

Reply:
Want to keep the fields returned same as `get_from_signal`

Reply:
For get_item though PROJECTION_EXPRESSION is just to filter out fields you don't want.